purchases/src/main/kotlin/com/revenuecat/purchases/galaxy/handler/ProductDataHandler.kt

```diff
@@ -0,0 +1,182 @@
+package com.revenuecat.purchases.galaxy.handler
+
+import android.os.Handler
+import com.revenuecat.purchases.ProductType
+import com.revenuecat.purchases.PurchasesError
+import com.revenuecat.purchases.PurchasesErrorCallback
+import com.revenuecat.purchases.PurchasesErrorCode
+import com.revenuecat.purchases.common.LogIntent
+import com.revenuecat.purchases.common.StoreProductsCallback
+import com.revenuecat.purchases.common.log
+import com.revenuecat.purchases.galaxy.GalaxyStrings
+import com.revenuecat.purchases.galaxy.IAPHelperProvider
+import com.revenuecat.purchases.galaxy.listener.ProductDataResponseListener
+import com.revenuecat.purchases.galaxy.toStoreProduct
+import com.revenuecat.purchases.galaxy.utils.isError
+import com.revenuecat.purchases.models.StoreProduct
+import com.samsung.android.sdk.iap.lib.vo.ErrorVo
+import com.samsung.android.sdk.iap.lib.vo.ProductVo
+import java.util.ArrayList
+
+internal class ProductDataHandler(
+    private val iapHelper: IAPHelperProvider,
+    private val mainHandler: Handler,
+) : ProductDataResponseListener {
+
+    companion object {
+        private const val GET_PRODUCT_DATA_TIMEOUT_MILLIS = 10_000L
+    }
+
+    @get:Synchronized
+    private var inFlightRequest: Request? = null
+
+    private data class Request(
+        val productIds: Set<String>,
+        val productType: ProductType,
+        val onReceive: StoreProductsCallback,
+        val onError: PurchasesErrorCallback,
+    )
+
+    @get:Synchronized
+    internal val productDataCache = mutableMapOf<String, ProductVo>()
```

Comment:
Just wondering, would it be worth caching the `StoreProduct` itself instead of `ProductVo`? So we don't need to transform it on every call to get product details. Might be missing a reason why we need to cache the original ProductVo though.